video/out/gpu/video.c

```diff
@@ -3376,15 +3376,12 @@ void gl_video_render_frame(struct gl_video *p, struct vo_frame *frame,
                     const struct ra_format *fmt = fbo->tex->params.format;
                     if (fmt->dummy_format)
                         fmt = p->fbo_format;
-
-                    bool r = ra_tex_resize(p->ra, p->log, &p->output_tex,
-                                           fbo->tex->params.w, fbo->tex->params.h,
-                                           fmt);
-                    if (r) {
-                        dest_fbo = &(struct ra_fbo) { p->output_tex };
-                        p->output_tex_valid = true;
-                    }
+                    r = ra_tex_resize(p->ra, p->log, &p->output_tex,
+                                      fbo->tex->params.w, fbo->tex->params.h,
+                                      fmt);
                 }
+                const struct ra_fbo *dest_fbo = r ? &(struct ra_fbo) { p->output_tex } : fbo;
```

Comment:
won't the lifetime of the inline struct end after this line?

Reply:
In C++ yes, in C no. In C it is alive until block ends.

Surly I'm the best person to tell this, lol.

Reply:
weird.

Reply:
No it's valid until the end of the if block. You can try it with and without `--video-sync=display-resample` to hit both cases.

Reply:
Was searching for a quote, because no one would believe me now.
https://port70.net/~nsz/c/c11/n1570.html#6.5.2.5p5
> The value of the compound literal is that of an unnamed object initialized by the initializer list. If the compound literal occurs outside the body of a function, the object has static storage duration; otherwise, it has automatic storage duration associated with the enclosing block.